src/alert-prelude.c

```diff
@@ -510,6 +611,160 @@ static int PacketToDataV6(const Packet *p, const PacketAlert *pa, idmef_alert_t
     SCReturnInt(0);
 }
 
+#ifdef HAVE_LIBJANSSON
+/**
+ * \brief Convert JSON object to Prelude additional data with
+ * the right type of data.
+ *
+ * \return 0 if ok
+ */
+int JsonToAdditionnalData(const char * key, json_t * value, idmef_alert_t *alert ) {
+    int ret;
+    const char *key_js;
+    json_t *value_js;
+    char local_key[128];
+    SCEnter();
+
+    if (!json_is_object(value) && key == NULL)
+        SCReturnInt(-1);
+
+    if (json_is_object(value)) {
+        json_object_foreach(value, key_js, value_js) {
+            snprintf(local_key, sizeof(local_key), "tls_%s", key_js);
+            ret = JsonToAdditionnalData(local_key, value_js, alert);
+        }
+    } else if (json_is_integer(value)) {
+        ret = AddIntData(alert,key, json_integer_value(value));
+    } else if (json_is_real(value)) {
+        ret = AddRealData(alert,key, json_real_value(value));
+    } else if (json_is_boolean(value)) {
+        ret = AddIntData(alert,key, json_is_true(value));
+    } else if (json_is_string(value)) {
+        ret = AddStringData(alert,key, json_string_value(value));
+    } else {
```

Comment:
The array case is handled here. Maybe it would be nice to treat it with specific code ?

Reply:
Done